### DISCUSSION TITLE
Nomad Network & Rasperry Pi issues.

### DISCUSSION THREAD

**Discussion by andymux** [markqvist/Reticulum#445] [Help & Questions]
I am trying to run Nomad Network on a Raspberry Pi 3.
The program has been installed correctly with Pip but it refuses to open the GUI when run from command line. The command prompt just appears again after running the program.

I have set the network interface in the reticulum config, in this case it's a serial LoRa device on ttyS0.

I am using 64 bit Raspian OS.

Any ideas?

**Comment by faragher** [comment]
That's very strange. The good news is I have a Pi3 running NomadNet right next to me, so it should be possible to fix.

So, there's no error messages, correct? And you're running either from a terminal or over ssh?

Can you post the logs from .nomadnetwork/logfile?

It there's no obvious issue, the second thing we should do is check to see if Reticulum is running well. Use

```
rnstatus
```

to check if it starts up and if the interfaces are functioning correctly. If something breaks there, you can increase the loglevel in .reticulum/config (by default). If that works perfectly, you can change the loglevel in .nomadnet/config.

You can also post your .reticulum/config and .nomadnetwork/config files if you think there may be an error.

**Reply by andymux** [reply]
Thanks for the reply! Great you have a Pi 3 working with Nomad.

Correct, there's no error messages and yes it's running in a terminal.

I'll post the log file below, there's a few errors!

rnstatus looks OK, it says the serial interface is up as per my serial configuration (9600bps)

Standby!

**Comment by andymux** [comment]
[logfile.txt](https://github.com/markqvist/Reticulum/files/14380325/logfile.txt)

**Reply by faragher** [reply]
Very strange. Good news is almost everything is working great and I found the big error.  Are you using a terminal emulator in a GUI? If so, could you try to use a different one?

The issue is

```
[2024-02-22 18:19:25] [Error] Type  : <class 'TypeError'>
[2024-02-22 18:19:25] [Error] Value : wcwidth() takes 1 positional argument but 2 were given
[2024-02-22 18:19:25] [Error] Trace : 
  File "/home/andy/.local/bin/nomadnet", line 8, in <module>
    sys.exit(main())
  File "/home/andy/.local/lib/python3.9/site-packages/nomadnet/nomadnet.py", line 51, in main
    program_setup(configarg, rnsconfigarg, daemon, console)
  File "/home/andy/.local/lib/python3.9/site-packages/nomadnet/nomadnet.py", line 11, in program_setup
    app = nomadnet.NomadNetworkApp(
 ...
  File "/home/andy/.local/lib/python3.9/site-packages/urwid/str_util.py", line 40, in get_char_width
    width = wcwidth.wcwidth(char, UNICODE_VERSION)
```

I'll do some digging, but it looks like something that would cause errors on other systems as well, so unless it's something specific in the terminal, I'll need to do some digging.

**Reply by faragher** [reply]
This appears to be an urwid problem, trying to call wcwidth with two arguments when it only accepts one (which may be a depreciated method or just an error).

What locale are you running? Do you have any unicode characters outside of Latin glyphs? (ie: Hiragana, katakana, kanji, possibly Cyrillic?)

I'm going to see if I can find a way around this temporarily for you.

**Reply by faragher** [reply]
I don't even have wcwidth installed.

You can try
```
pip install urwid==2.1.2
```

and see if their latest updates caused the problems. I have to stress it's a temporary solution, and I'm concerned that this will be a major issue for new installs, but it's worth a shot. Going to update now and see if I can replicate your issue.

**Reply by faragher** [reply]
Strangely urwid 2.6.4 doesn't cause that problem for me. I suggest making sure you set up your locale correctly, and if that doesn't fix the problem, we'll keep digging.

**Reply by faragher** [reply]
Reference https://github.com/urwid/urwid/pull/803

It appears to have been an update in the last week. Still not sure why it affects you but not me, unless my pip repos are stale.

**Reply by andymux** [reply]
Thanks for this, it is strange.

I'll try 
`pip install urwid==2.1.2`

This is a fresh Raspian image. Initially I had trouble installing nomadnetwork 
I had this error when trying to install:

https://stackoverflow.com/questions/73830524/attributeerror-module-lib-has-no-attribute-x509-v-flag-cb-issuer-check

So I solved it with:

`sudo apt remove python3-pip`
`wget https://bootstrap.pypa.io/get-pip.py`
`sudo python3 get-pip.py`

And then after a reboot:
`pip install pyopenssl --upgrade`

I'm also using English language, no unique characters.

**Reply by andymux** [reply]
It's running! Awesome!

Thanks so much 👍🏻

**Reply by faragher** [reply]
That SSL error drives me nuts every time I've installed a Pi since . . . Buster? Bullseye? Anyway, someday I'll remember how I fix it every time. :D

Glad to hear it's working! Have fun!

**Reply by andymux** [reply]
Haha yeah! It's a pain and will be a roadblock for other users getting on this definately.
